### PR TITLE
Motion blur schema update and multisegment deformation motion blur

### PR DESCRIPTION
**Pull request by bareya**
* new schema for deformation and transformation motion blur
* by default motion blur is on
* transformation samples set to 3
* deformation samples set to 3
* individual control over motion blur, deformation, transformation samples per object

### CODE REVIEW

File: plugin/hdCycles/basisCurves.cpp [hunk 1]
```diff
@@ -180,28 +169,39 @@ HdCyclesBasisCurves::_PopulateCurveMesh(HdRenderParam* renderParam)
...
 
+    m_cyclesHair->use_motion_blur = true;
+    m_cyclesHair->motion_steps = static_cast<unsigned int>(numSamples + ((numSamples % 2) ? 0 : 1));
+
+    attr_mP = attributes->add(ccl::ATTR_STD_MOTION_VERTEX_POSITION);
     ccl::float3* mP = attr_mP->data_float3();
-    for (size_t i = 0; i < m_pointSamples.count; ++i) {
-        if (m_pointSamples.times.data()[i] == 0.0f) {
+
+    for (size_t i = 0; i < motion_samples.count; ++i) {
```
Comment (dedoardo) [inline plugin/hdCycles/basisCurves.cpp:199]: Could this go through the Attribute source and be resampled like transforms?

Comment (dedoardo) [reply plugin/hdCycles/basisCurves.cpp:199]: Ah, saw the todo in the mesh class :+1:

**Comment by boberfly** [comment]
Visually LGTM but I don't have time currently to test thorough. I am wondering though if Hydra is smart enough to not make everything deform motion blur of 3 steps if it's a static non-changing geo and we now default to it. Cheers!

**Comment by bareya** [comment]
@boberfly In `Hydra` we receive samples, we don't request them from Scene Delegate. You can request 3 transformation samples, but SamplePrimvar is going to receive only 1 and then there is no motion blur.

File: plugin/hdCycles/mesh.cpp [hunk 2]
```diff
@@ -615,30 +608,35 @@ HdCyclesMesh::_PopulateMotion(HdSceneDelegate* sceneDelegate, const SdfPath& id)
...
+    auto& times = motion_samples.times;
+    auto& values = motion_samples.values;
+
+    ccl::AttributeSet* attributes = &m_cyclesMesh->attributes;
+    ccl::Attribute* attr_mP = attributes->find(ccl::ATTR_STD_MOTION_VERTEX_POSITION);
+    if (attr_mP) {
+        attributes->remove(attr_mP);
+    }
 
     if (numSamples <= 1) {
```
Comment (bareya) [inline plugin/hdCycles/mesh.cpp:625]: @boberfly here we terminate deformation motion blur population, if number of samples is insufficient.

**Pull request merged**